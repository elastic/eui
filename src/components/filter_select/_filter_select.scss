.euiFilterSelect__items {
  @include euiScrollBar;

  overflow-y: auto;
  width: $euiSize * 20;
  max-height: $euiSize * 30;
}

.euiFilterSelect__note {
  height: $euiSize * 4;
  text-align: center;
  display: flex;
  align-items: center;
  justify-content: space-around;
}

.euiFilterSelect__noteContent {
  color: $euiColorDarkShade;
  font-size: $euiFontSizeS;
}
