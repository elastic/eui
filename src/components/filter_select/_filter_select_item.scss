.euiFilterSelectItem {
  @include euiFontSizeS;

  padding: $euiSizeXS $euiSizeM;
  width: 100%;
  text-align: left;
  color: $euiTextColor;
  border-bottom: $euiBorderThin;
  border-color: darken($euiColorLightestShade, 2%);

  &:hover, &:focus {
    text-decoration: underline;
  }

  &:focus {
    background-color: $euiFocusBackgroundColor;
  }

  &.euiContextMenuItem-isDisabled {
    color: $euiButtonColorDisabled;
    cursor: default;

    &:hover, &:focus {
      text-decoration: none;
    }
  }
}
