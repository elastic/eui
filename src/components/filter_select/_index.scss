@import 'filter_select';
@import 'filter_select_item';
