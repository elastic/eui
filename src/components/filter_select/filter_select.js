import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

export const EuiFilterSelect = ({
  children,
  className,
  ...rest,
}) => {
  const classes = classNames('euiFilterSelect', className);

  return (
    <div
      className={classes}
      {...rest}
    >
      {children}
    </div>
  );
};

EuiFilterSelect.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
};
