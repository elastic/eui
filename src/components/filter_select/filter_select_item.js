import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import {
  EuiFlexGroup,
  EuiFlexItem,
} from '../flex';

import {
  EuiIcon,
} from '../icon';

export const EuiFilterSelectItem = ({
  children,
  className,
  disabled,
  isSelected,
  ...rest,
}) => {
  const classes = classNames('euiFilterSelectItem', className);

  return (
    <button
      className={classes}
      type="button"
      disabled={disabled}
      {...rest}
    >
      <EuiFlexGroup
        alignItems="center"
        gutterSize="s"
        component="span"
      >
        <EuiFlexItem grow={false}>
          <EuiIcon type={isSelected ? 'checkInCircleFilled' : 'empty'} />
        </EuiFlexItem>
        <EuiFlexItem>
          {children}
        </EuiFlexItem>
      </EuiFlexGroup>
    </button>
  );
};

EuiFilterSelectItem.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
};
