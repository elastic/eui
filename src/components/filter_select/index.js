export {
  EuiFilterSelect,
} from './filter_select';

export {
  EuiFilterSelectItem,
} from './filter_select_item';
