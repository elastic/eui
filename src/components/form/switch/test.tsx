import React from 'react';

import { EuiSwitch } from './switch';

export const Test = ({ checked }: { checked: boolean }) => (
  <EuiSwitch
    checked={checked}
    label="Test"
    onChange={e => console.log(e.target.checked)}
  />
);
