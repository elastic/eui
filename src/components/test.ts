export function add(x: number, y: number) {
  return x + y;
}

add(5, 4);
