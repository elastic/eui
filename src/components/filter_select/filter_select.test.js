import React from 'react';
import { render } from 'enzyme';
import { requiredProps } from '../../test';

import { EuiFilterSelect } from './filter_select';

describe('EuiFilterSelect', () => {
  test('is rendered', () => {
    const component = render(
      <EuiFilterSelect {...requiredProps} />
    );

    expect(component)
      .toMatchSnapshot();
  });
});
