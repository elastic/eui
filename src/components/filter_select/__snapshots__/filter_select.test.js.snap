// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EuiFilterSelect is rendered 1`] = `
<div
  aria-label="aria-label"
  class="euiFilterSelect testClass1 testClass2"
  data-test-subj="test subject string"
/>
`;
